WordPress/Classes/ViewRelated/Post/WPPostViewController.m

```diff
@@ -1445,7 +1445,7 @@ - (void)savePost
 
     [self.view endEditing:YES];
     
-    if (!self.post.isScheduled && [self.post.original.status isEqualToString:PostStatusDraft]  && [self.post.status isEqualToString:PostStatusPublish]) {
+    if (!self.post.hasFuturePublishDate && [self.post.original.status isEqualToString:PostStatusDraft]  && [self.post.status isEqualToString:PostStatusPublish]) {
```

Comment:
I think we could improve readability a bit here if instead of listing the conditions inside the if block we put them inside a separate method of its own with a clearer name.

One example of what I'm thinking of:

```objective-c
if ([self postMustBePublishedNow])
{
    // code
}
```

With this approach we'd state the intent of the check clearly - regardless of if the particular conditions need to be changed or adapted in the future.

The method to add would be something along the lines of:

```objective-c
- (BOOL)postMustBePublishedNow
{
    return !self.post.hasFuturePublishDate && [self.post.original.status isEqualToString:PostStatusDraft]  && [self.post.status isEqualToString:PostStatusPublish];
}
```